src/pkrd/rng/mt.rs

```diff
@@ -0,0 +1,77 @@
+pub struct MT {
+    index: usize,
+    mt: [u32; 624],
+}
+
+impl MT {
+    pub fn new(seed: u32) -> MT {
+        let mut rng = MT {
+            mt: [0; 624],
+            index: 624,
+        };
+        rng.init(seed);
+
+        rng
+    }
+
+    fn init(&mut self, seed: u32) {
+        self.mt[0] = seed;
+
+        let mut seed = seed;
+        for i in 1..624 {
+            seed = 0x6c078965u32.wrapping_mul(seed ^ (seed >> 30)).wrapping_add(i);
+            self.mt[i as usize] = seed;
+        }
+    }
+
+    pub fn next(&mut self) -> u32 {
+        if self.index == 624 {
+            self.shuffle();
+        }
+
+        let y = self.mt[self.index];
+        self.index += 1;
+
+        // Game doesn't store tempered in memory
+        //y ^= y >> 11;
+        //y ^= (y << 7) & 0x9d2c5680;
+        //y ^= (y << 15) & 0xefc60000;
+        //y ^= y >> 18;
+        y
+    }
+
+    fn shuffle(&mut self) {
+        for i in 0..227 {
+            let y = (self.mt[i] & 0x80000000) | (self.mt[i+1] & 0x7fffffff);
+
+            let mut y1 = y >> 1;
+            if (y & 1) == 1 {
+                y1 ^= 0x9908b0df;
```

Comment:
I just realized, it looks like we don't use the result of `y1` here or in the loop below.  Is this supposed to be `y`?

Reply:
Typo, y1 is actually used